### PR TITLE
leafnode: 20121101 -> 20140727; leafnode1: init at 1.12

### PR DESCRIPTION
**Pull request by ghost**
sn: init at 0.38

## Description of changes

- Leafnode, 2.x development version. Update to 20140727. [Changelog](http://krusty.dt.e-technik.tu-dortmund.de/~ma/leafnode/beta/ChangeLog.txt)

- Leafnode, 1.x stable release, an easy to use NNTP news server for small networks. https://leafnode.sourceforge.io/status.shtml

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ne9z** [comment]
@jwiegley As the original author of this package definition, you might want to review this update.  

I also did #248320 in an effort to bring more Usenet (Netnews/NNTP) support into NixOS, which I plan to use with Emacs/Gnus.

**Comment by jwiegley** [comment]
Wow, I haven't used leafnode since I was a young dinosaur. I am no longer equipped to review!

File: pkgs/servers/news/leafnode1/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ lib, stdenv, fetchurl, pcre2, libxcrypt }:
+
+let
+  pname = "leafnode";
+  version = "1.12.0";
+in stdenv.mkDerivation {
```
Comment (AndersonTorres) [inline pkgs/servers/news/leafnode1/default.nix:6]: Instead of let-against-rec, use

```suggestion
stdenv.mkDerivation (finalAttrs: {
  pname = "leafnode";
  version = "1.12.0";
```

File: pkgs/servers/news/leafnode1/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  inherit pname version;
+
+  src = fetchurl {
+    url =
+      "https://downloads.sourceforge.net/project/leafnode/leafnode/${version}/leafnode-${version}.tar.gz";
+    sha256 = "sha256-tGfOcyH2F6IeglfY00u199eKusnn6HeqD7or3Oz3ed4=";
+  };
```
Comment (AndersonTorres) [inline pkgs/servers/news/leafnode1/default.nix:14]:
```suggestion
  src = fetchurl {
    url = "https://downloads.sourceforge.net/project/leafnode/leafnode/${version}/leafnode-${finalAttrs.version}.tar.gz";
    hash = "sha256-tGfOcyH2F6IeglfY00u199eKusnn6HeqD7or3Oz3ed4=";
  };
```

File: pkgs/servers/news/leafnode1/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  configureFlags = [
+    "--with-ipv6"
+  ];
+
+  buildInputs = [ pcre2 ];
+
+  meta = {
+    homepage = "https://leafnode.sourceforge.io/index.shtml";
+    description =
+      "Implementation of a store & forward NNTP proxy, stable release";
```
Comment (AndersonTorres) [inline pkgs/servers/news/leafnode1/default.nix:25]:
```suggestion
    description = "Implementation of a store & forward NNTP proxy, stable release";
```

File: pkgs/servers/news/sn/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  buildInputs = [ zlib ucspi-tcp ];
+
+  meta = {
+    homepage = "https://patrik.iki.fi/sn/";
+    description = "A small news system for small sites";
+    license = lib.licenses.gpl2Only;
+    platforms = lib.platforms.unix;
+    maintainers = [ lib.maintainers.ne9z ];
+  };
+}
```
Comment (AndersonTorres) [inline pkgs/servers/news/sn/default.nix:33]:
```suggestion
{ lib
, stdenv
, fetchurl
, ucspi-tcp 
, zlib
}:

stdenv.mkDerivation (finalAttrs: {
  pname = "sn";
  version = "0.3.8";
  
  src = fetchurl {
    url = "https://www.ibiblio.org/pub/Linux/system/news/transport/sn-${finalAttrs.version}.tar.bz2";
    hash = "sha256-jOfHfVh4FJk8on1bvyu4sq0TERmVthE9yRdIRZbb3fA=";
  };

  postPatch = ''
    substituteInPlace Makefile \
      --replace  '/usr/local' '${placeholder "out"}'
  '';

  buildInputs = [ 
    ucspi-tcp
    zlib
  ];
  
  preInstall = ''
    mkdir -p $out/sbin $out/man/man8
  '';

  meta = {
    homepage = "https://patrik.iki.fi/sn/";
    description = "A small news system for small sites";
    license = lib.licenses.gpl2Only;
    maintainers = [ lib.maintainers.ne9z ];
    platforms = lib.platforms.unix;
  };
})
```

And put this in another PR.

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -26367,6 +26367,8 @@ with pkgs;
 
   leafnode = callPackage ../servers/news/leafnode { };
 
+  leafnode1 = callPackage ../servers/news/leafnode1 { };
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:26370]: Put this in `leafnode/1.nix`

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by ne9z** [comment]
@AndersonTorres  Hello, I've made the requested changes.  Thanks for the detailed review!

**Review by AndersonTorres — APPROVED** [review]

**Comment by AndersonTorres** [comment]
Ping me 36h to merge

**Comment by ne9z** [comment]
@AndersonTorres Hello.  Yes, I have opened a separate pull request for the `sn` package, see #248971.  Thank you!
